From 757f828e0876ed99c22d355b0fc44a5c4a7814e5 Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@rvc.renesas.com>
Date: Wed, 13 Mar 2019 16:06:58 +0700
Subject: [PATCH] Demo3D: Change detect camera input

Since we have many channel of VIN, the previous way to detect camera
is not success.
Change the way to get camera input and use the first camera.

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@rvc.renesas.com>
---
 Demo3D.c | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/Demo3D.c b/Demo3D.c
index 5457b12..5cea09b 100644
--- a/Demo3D.c
+++ b/Demo3D.c
@@ -1270,17 +1270,20 @@ char *camera_get_device()
   while(1){
     char device[20];
     int fd;
+    char data;
     sprintf (device, "/dev/video%d", i);
     if((fd = open(device, O_RDONLY)) == -1){              
         break;
-    }          
-    struct v4l2_fmtdesc v4l2_fmtdesc;
-    v4l2_fmtdesc.index = 0;
-    v4l2_fmtdesc.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
-    
-    if (ioctl(fd, VIDIOC_ENUM_FMT , &v4l2_fmtdesc) != -1){        
-      sprintf (cam_dev, "/dev/video%d", i);
+    }   
+
+    //Check if camera is available      
+    if(read(fd,&data,sizeof(data)) != -1){ 
+	sprintf (cam_dev, "/dev/video%d", i);
+	//Using this camera and break the loop
+ 	close(fd);
+	break;
     }
+
     close(fd);
     i++;
   }  
-- 
2.7.4

